core/src/main/java/io/github/thibaultbee/streampack/streamers/bases/BaseStreamer.kt

```diff
@@ -143,36 +148,25 @@ abstract class BaseStreamer(
      * Manages error on stream.
      * Stops only stream.
      *
-     * @param error triggered [StreamPackError]
+     * @param e triggered [Exception]
      */
-    private fun onStreamError(error: StreamPackError) {
+    private fun onStreamError(e: Exception) {
         try {
             runBlocking {
                 stopStream()
             }
         } catch (e: Exception) {
             Logger.e(TAG, "onStreamError: Can't stop stream")
         } finally {
-            onErrorListener?.onError(error)
+            onErrorListener?.onError(e)
         }
     }
 
-    protected var audioEncoder = if (audioSource != null) {
-        AudioMediaCodecEncoder(audioEncoderListener, onInternalErrorListener)
-    } else {
-        null
-    }
-    protected var videoEncoder = if (videoSource != null) {
-        VideoMediaCodecEncoder(
-            videoEncoderListener,
-            onInternalErrorListener,
-            videoSource.hasSurface,
-            sourceOrientationProvider
-        )
-    } else {
-        null
-    }
-    override val settings = BaseStreamerSettings(audioSource, audioEncoder, videoEncoder)
+    protected var audioEncoder: MediaCodecEncoder? = null
+    protected var videoEncoder: MediaCodecEncoder? = null
+    protected var codecSurface: CodecSurface? = null
```

Comment:
tbh I don't see the benefit of breaking the codecsurface out to its own thing as opposed to keeping it tucked away inside the video encoder. Now its just one more thing that needs to get started/stopped/released

Reply:
Encoder are now agnostic of the fact it is a audio or video encoder.
And `CodecSurface` will become a powerful element that is not related to a single encoder.

It is not that difficult to call start/stop/release.